Makefile

```diff
@@ -1,174 +1,296 @@
+REGULAR_DATA_ASM_SRCS := $(filter-out $(DATA_ASM_SUBDIR)/maps.s $(DATA_ASM_SUBDIR)/map_events.s, $(wildcard $(DATA_ASM_SUBDIR)/*.s))
+
 DATA_ASM_SRCS := $(wildcard $(DATA_ASM_SUBDIR)/*.s)
 DATA_ASM_OBJS := $(patsubst $(DATA_ASM_SUBDIR)/%.s,$(DATA_ASM_BUILDDIR)/%.o,$(DATA_ASM_SRCS))
 
-SONG_SRCS := $(wildcard $(SONG_SUBDIR)/*.s)
-SONG_OBJS := $(patsubst $(SONG_SUBDIR)/%.s,$(SONG_BUILDDIR)/%.o,$(SONG_SRCS))
-
-MID_SRCS := $(wildcard $(MID_SUBDIR)/*.mid)
-MID_OBJS := $(patsubst $(MID_SUBDIR)/%.mid,$(MID_BUILDDIR)/%.o,$(MID_SRCS))
-
-OBJS := $(C_OBJS) $(ASM_OBJS) $(DATA_ASM_OBJS) $(SONG_OBJS) $(MID_OBJS)
+OBJS     := $(C_OBJS) $(C_ASM_OBJS) $(ASM_OBJS) $(DATA_ASM_OBJS) $(SONG_OBJS) $(MID_OBJS)
 OBJS_REL := $(patsubst $(OBJ_DIR)/%,%,$(OBJS))
 
-TOOLDIRS := $(filter-out tools/agbcc tools/binutils,$(wildcard tools/*))
-TOOLBASE = $(TOOLDIRS:tools/%=%)
-TOOLS = $(foreach tool,$(TOOLBASE),tools/$(tool)/$(tool)$(EXE))
-
-.PHONY: all rom tools clean-tools mostlyclean clean compare tidy speed_eaters unk_8D94B9C unk_8E1FE28 unk_8E8490C $(TOOLDIRS)
-
-MAKEFLAGS += --no-print-directory
-
-AUTO_GEN_TARGETS :=
+SUBDIRS  := $(sort $(dir $(OBJS)))
+$(shell mkdir -p $(SUBDIRS))
 
-all: tools rom
+# Pretend rules that are actually flags defer to `make all`
+modern: all
+compare: all
 
+# Other rules
 rom: $(ROM)
 ifeq ($(COMPARE),1)
-	@$(SHA1) katam.sha1
+	@$(MAKE) -C $(SUBGAME_LOADERS) compare
+	@$(MAKE) -C multi_boot/unk_8D94B9C compare
+	@$(MAKE) -C multi_boot/unk_8E1FE28 compare
+	@$(MAKE) -C multi_boot/unk_8E8490C compare
+	@$(SHA1) $(FILE_NAME).sha1
 endif
 
-tools: $(TOOLDIRS)
-
-$(TOOLDIRS):
-	@$(MAKE) -C $@
-
-# For contributors to make sure a change didn't affect the contents of the ROM.
-compare: all
+syms: $(SYM)
 
 SUBGAME_LOADERS := multi_boot/subgame_loaders
 
-mostlyclean: tidy
-	@rm -f sound/direct_sound_samples/*.bin
-	@rm -f $(SONG_OBJS) $(MID_SUBDIR)/*.s
-	@find . \( -iname '*.1bpp' -o -iname '*.4bpp' -o -iname '*.8bpp' -o -iname '*.gbapal' -o -iname '*.lz' -o -iname '*.rl' -o -iname '*.latfont' -o -iname '*.hwjpnfont' -o -iname '*.fwjpnfont' \) -exec rm {} +
-	@rm -f $(AUTO_GEN_TARGETS)
+clean: tidy clean-tools clean-generated clean-assets
 	@$(MAKE) -C $(SUBGAME_LOADERS) $@
 	@$(MAKE) -C multi_boot/unk_8D94B9C $@
 	@$(MAKE) -C multi_boot/unk_8E1FE28 $@
 	@$(MAKE) -C multi_boot/unk_8E8490C $@
 
-clean-tools:
-	@$(foreach tooldir,$(TOOLDIRS),$(MAKE) clean -C $(tooldir);)
-
-clean: mostlyclean clean-tools
+clean-assets:
+	rm -f $(MID_SUBDIR)/*.s
+	rm -f sound/direct_sound_samples/*.bin
+	rm -f $(DATA_ASM_SUBDIR)/layouts/layouts.inc $(DATA_ASM_SUBDIR)/layouts/layouts_table.inc
+	rm -f $(DATA_ASM_SUBDIR)/maps/connections.inc $(DATA_ASM_SUBDIR)/maps/events.inc $(DATA_ASM_SUBDIR)/maps/groups.inc $(DATA_ASM_SUBDIR)/maps/headers.inc
+	find . \( -iname '*.1bpp' -o -iname '*.4bpp' -o -iname '*.8bpp' -o -iname '*.gbapal' -o -iname '*.lz' -o -iname '*.rl' -o -iname '*.latfont' -o -iname '*.hwjpnfont' -o -iname '*.fwjpnfont' \) -exec rm {} +
```

Comment:
It'd be better if we prepend `@` to all these lines. Currently there are way too many output lines when you run `make clean` considering there are subdirectories. And they're not helpful at all.

Same for other lines starting with `rm` and those in subdirectories.
(But brief and informative ones can still be there, like `@echo "rm -f <AUTO_GEN_TARGETS>"`)

Reply:
Done. Though the way I did it might be a bit opinionated.